app/javascript/components/payment-error-button.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <div class="payment-error-button">
+    <modal name="apology-modal">
+      <div class="invoice__apology-modal-text">
+        <h1>Error en el pago</h1>
+        Disculpe las molestias :(, lo vamos a revisar para que puedas pagar en el futuro
+      </div>
+    </modal>
+    <div class="invoice__error-btn ">
+        <button
+          type="button" v-if="showApologyButton"
+          class="btn"
+          @click="show"
+        >
+          ¿No funciona?
+        </button>
+    </div>
+  </div>
+</template>
+
+<script>
+
+import { mapState, mapActions, mapGetters } from 'vuex';
+import invoiceApi from '../api/invoices';
+const SHOW_APOLOGY_BUTTON = 10000;
+const CLOSE_MODAL_AFTER_IDLE_WAIT = 7000;
+
+export default {
+  data() {
+    return {
+      showApologyButton: false,
+    };
+  },
+  methods: {
+    show() {
+      this.$modal.show('apology-modal');
+      invoiceApi.notifyPaymentError();
+      setTimeout(() => {
+        this.$modal.hide('apology-modal');
+      }, CLOSE_MODAL_AFTER_IDLE_WAIT);
+    },
+    hide() {
+      this.$modal.hide('apology-modal');
+    },
+  },
+  mounted() {
+    setTimeout(() => {
+      this.showApologyButton = true;
+    }, SHOW_APOLOGY_BUTTON);
+  },
+}
+</script>
```

Comment:
Newline required at end of file but not found                              eol-last